substrate/frame/executive/src/lib.rs

```diff
@@ -687,13 +665,45 @@ where
 	}
 
 	/// Execute given extrinsics.
-	fn apply_extrinsics(extrinsics: impl Iterator<Item = Block::Extrinsic>) {
-		extrinsics.into_iter().for_each(|e| {
-			if let Err(e) = Self::apply_extrinsic(e) {
-				let err: &'static str = e.into();
-				panic!("{}", err)
+	fn apply_extrinsics<F>(
+		mode: ExtrinsicInclusionMode,
+		extrinsics: impl Iterator<Item = Block::Extrinsic>,
+		mut f: F,
+	) -> Result<(), ExecutiveError>
+	where
+		F: FnMut(Block::Extrinsic, bool) -> ApplyExtrinsicResult,
+	{
+		let mut last_inherent_idx = 0;
```

Comment:
```suggestion
		let mut first_non_inherent_idx = 0;
```

Is kind of more correct.